### PR TITLE
simpler way to create a new catalog

### LINKED ISSUE #11: Simplify ProjectCatalog.create()
### Description

Currently, if I want to rerun a workflow without overwriting the ProjectCatalog, I need something like:

```
if os.path.isfile(catalog_path):
   pcat = ProjectCatalog(catalog_path)
else:
   pcat = ProjectCatalog.create(catalog_path, project=project, overwrite=True)
```

I think we could instead have a one-liner like:

`pcat = ProjectCatalog(catalog_path, project=project, create_if_required=True, overwrite=False)`

### PR DESCRIPTION
**Pull request by RondeauG**
### Pull Request Checklist:
- [x] `pre-commit` hooks are installed/active in my local clone (`$ pre-commit install`)
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #11
- [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] If a merge request has been made in parallel to this PR in xscen-notebooks, it is merged and the submodules have been updated.
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Adds a `create` argument to `ProjectCatalog()`:
  * `ProjectCatalog("path.json", create=True, project=project, overwrite=False)` (default) will only create a new catalog if one doesn't already exist. If it exists, it will drop the creation-specific kwargs and open the JSON normally.
  *  `ProjectCatalog("path.json", create=True, project=project, overwrite=True)` will create a new catalog, overwriting if necessary.
  * Any version without `create` will work as previously and crash if the file doesn't exist.

### Does this PR introduce a breaking change?

No.

### CODE REVIEW

File: xscen/catalog.py [hunk 1]
```diff
@@ -411,6 +411,14 @@ def create(
...
 
     def __init__(self, df, *args, **kwargs):
+        if "create" in kwargs:
+            kwargs.pop("create")
+            if isinstance(df, (str, Path)) and (
+                not os.path.isfile(Path(df)) or kwargs.get("overwrite", None) is True
+            ):
+                self.create(df, **kwargs)
+        kwargs.pop("project", None)
+        kwargs.pop("overwrite", None)
```
Comment (aulemahal) [inline xscen/catalog.py:429]:
```suggestion
    def __init__(self, df, *args, create=False, overwrite=None, project=None, **kwargs):
        if create:
            if isinstance(df, (str, Path)) and (not os.path.isfile(Path(df)) or overwrite):
                self.create(df, project=project, overwrite=overwrite)
```
If kwargs are not passed directly to a parent, I find it makes more sense to have them explicitly in the signature.

But in fact I am wondering if your signatures make sense at all here. When I dig, I can't see what `*args` could be and the only meaningful `kwarg` are `read_csv_kwargs` and `registry`. Why not list them directly? I can push something if you agree.

Comment (aulemahal) [reply xscen/catalog.py:429]: (outdated comment, but I think it addresses your last changes too)

Comment (RondeauG) [reply xscen/catalog.py:429]: I'm honestly not sure either... I think we might just have used `*args, **kwargs` as a default way to support whatever changes could occur in `intake-esm`?

Comment (aulemahal) [reply xscen/catalog.py]: Indeed, it's a good idea to be ready for future changes... In that case, then I think this is satisfactory!

**Review by aulemahal — APPROVED** [review]

**Pull request merged**